platform/ios/src/MGLMapView.mm

```diff
@@ -680,6 +696,45 @@ - (void)reachabilityChanged:(NSNotification *)notification
     _isWaitingForRedundantReachableNotification = NO;
 }
 
+// This is similar to macOS' `restorableStateKeyPaths` used for state restoration
+// Support for `UIStateRestoring` is still TBD: see https://github.com/mapbox/mapbox-gl-native/issues/3660
++ (NSArray *)mglRestorableStateKeyPaths {
+    return @[@"camera", @"debugMask", @"styleURL"];
+}
+
+- (id)valueAtTerminationForKey:(NSString*)key ofClass:(Class)classType {
```

Comment:
> #13120 is fixed by the other parts of this PR that explicitly check for an _mbglMap, correct?

Yes, though as you and @ChrisLoer say there are other parts of the code base that need tearing down, so perhaps it would be more accurate to say "mostly fixed".